### ISSUE TITLE
'Query.text' does not update text if element has been already added to document

### ISSUE THREAD

**Issue by amoilanen** [clientside/amplesdk#149]
With the old version of Ample both test cases pass, with the newest one the last test case fails:

``` javascript

module("Query.text");

(function() {

    //Passes
    test("Changing text node value before adding it to document", function() {
        var el = ample.query("<div name=\"test-before-insert\"/>").text("test_text_before")[0];
        ample.documentElement.appendChild(el);

        var domEl = document.querySelector("div[name=test-before-insert]");

        equal(el.$getContainer().innerHTML, "test_text_before");
        equal(domEl.innerHTML, "test_text_before");
    });

    //Fails
    test("Changing text node value after adding it to document", function() {
        var el = ample.query("<div name=\"test-after-insert\"/>").appendTo(ample.documentElement);

        el.text("test_text_after");
        el = el[0];

        var domEl = document.querySelector("div[name=test-after-insert]");

        equal(el.$getContainer().innerHTML, "test_text_after");
        equal(domEl.innerHTML, "test_text_after");
    });
})();
```

**Comment by amoilanen** [comment]
I can attach a patch later but here are the changes in ample/runtime/modules/manipulation.js that seem to fix the problem. It may cause the method to execute a bit slower:

``` javascript
// Text (get/set)
cQuery.prototype.text   = function(sValue) {
//->Guard
    fGuard(arguments, [
        ["value",   cObject, true]
    ]);
//<-Guard

    if (arguments.length > 0) {
        // Replace children with a text node
        fQuery_each(this, function() {
            var oText   = null;
            if (this.childNodes.length == 1 && (this.firstChild.nodeType == 3 || this.firstChild.nodeType == 4))
                fElement_removeChild(this, this.firstChild);
            else {
                while (this.lastChild)
                    fElement_removeChild(this, this.lastChild);
            }
            // Add child
            oText = fElement_appendChild(this, fDocument_createTextNode(this.ownerDocument, cString(sValue)));
        });
        return this;
    }
    else {
        // Get inner text
        var aText   = [];
        fQuery_each(this, function(){
            aText.push(fNode_getTextContent(this));
        });
        return aText.join('');
    }
};
```

**Comment by ilinsky** [comment]
Fixed per rev 234de5e59dc525601ce6170fd9b9f78acb651c0c. 
Test case added at https://github.com/clientside/amplesdk-tests/tree/master/manual/issues/149/
